graylog2-server/src/main/java/org/graylog2/rest/resources/streams/StreamResource.java

```diff
@@ -378,6 +381,13 @@ public void delete(@ApiParam(name = "streamId", required = true) @PathParam("str
     private Stream deleteInner(String streamId, UserContext userContext) throws NotFoundException {
         checkPermission(RestPermissions.STREAMS_EDIT, streamId);
         checkNotEditableStream(streamId, "The stream cannot be deleted.");
+        streamDeletionGuards.forEach(guard -> {
```

Comment:
Let's move the stream guard checks into the `StreamService` to ensure we apply the guards for all stream deletions, not only the deletions via HTTP API.